src/App.tsx

```diff
@@ -1,223 +1,185 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { deleteTodo, getTodos, updatingTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorMasage } from './components/ErrorMessage/ErrorMessage';
+import { Loader } from './components/Loader/Loader';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [countOfItemsLeft, setCountOfItemsLeft] = useState(0);
+  const [filterType, setFilterType] = useState('All');
+  const [isAllSelected, setIsAllSelected] = useState(false);
+
+  const uploadTodos = useCallback(
+    async () => {
+      setIsLoading(true);
+      try {
+        const data = await getTodos(user?.id);
+
+        setTodos(data);
+        setVisibleTodos(data);
+      } catch (err) {
+        setErrorMessage('upload a todo');
+      } finally {
+        setIsLoading(false);
+      }
+    }, [user],
+  );
+
+  const selectAllTodos = () => {
+    const isAllSelectedNow = visibleTodos.every(todo => todo.completed);
+
+    setIsAllSelected(isAllSelectedNow);
+
+    const selectedTodos = visibleTodos.map((todo) => {
+      const { completed, id } = todo;
+
+      if (isAllSelectedNow) {
+        return { ...todo, completed: false };
+      }
+
+      if (completed) {
+        return todo;
+      }
+
+      updatingTodo(id, true);
+
+      return { ...todo, completed: true };
+    });
+
+    setVisibleTodos(selectedTodos);
+  };
+
+  const clearCompleted = () => {
+    const filteredTodos = visibleTodos.filter((todo) => {
+      const { completed, id } = todo;
+
+      if (completed) {
+        deleteTodo(id);
+      }
+
+      return !completed;
+    });
```

Comment:
I'm not sure that it works correctly if the server return error on delete. Looks like on db todos will be, but on local you delete it